### ISSUE TITLE
diagnosticSeverityOverrides is not overriding properly when there is also pyrightconfig.json

### ISSUE THREAD

**Issue by tonyzhang9090** [microsoft/pyright#3039]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
When the VS Code project root has a `pyrightconfig.json` (for configs like “include”), setting `python.analysis.diagnosticSeverityOverrides` in settings.json doesn’t work as expected.

I think it’s caused by diagnosticSeverityOverrides logic in [configOptions.ts](https://github.com/microsoft/pyright/blob/b740ddfae5e6e74fcc819a1c8a78f83bc2571309/packages/pyright-internal/src/common/configOptions.ts#L899-L901) is incorrectly override by the the pyrightconfig config file, even when the config file doesn’t contain any diagnostic settings (like “reportMissingImports”).

**To Reproduce**
Any pyrightconfig.json, just an example below:
```json
{
  "include": ["**/*.py"]
}
```

VS code settings.json
```json
{
    "python.analysis.diagnosticSeverityOverrides": { 
        "reportMissingImports" : "none",
    }
}
```

`reportMissingImports` lints are still reported in VS Code.

**Expected behavior**
`python.analysis.diagnosticSeverityOverrides` should be working properly even when there is pyrightconfig.json in project root.

**VS Code extension or command-line**
VS Code version 1.64

**Comment by erictraut** [comment]
Thanks for the bug report. I agree that the current behavior is wrong. This is a long-standing bug, a couple of years old. I'm surprised no one has reported it previously.

**Comment by erictraut** [comment]
This is included in pyright 1.1.222, which I just published. It will also be included in the next release of pylance.
